### DISCUSSION TITLE
How to create a list of InfiniteQuery

### DISCUSSION THREAD

**Discussion by tchaton** [TanStack/query#5007] [General]
Dear React Query Team,

I have tried to implement a multiple InfiniteQuery but I am struggling. I tried using useQueries and array mapping but didn't manage to write something well. 

I need some of the infinite queries to refetch automatically if a list of ids changes. Each infinite query is associated to an id.

Any examples or advices on how to do so?

**Comment by tchaton** [comment]
I tried the following but it doesn't work:

```js
export function useListMultipleSourceArtifacts(ids?: string[], projectId?: string) {
  const queries = ids!.map(id => {
    return { 
      queryKey: ['useListArtifacts', id],
      queryFn: async () => UseListSourceArtifacts(id, projectId)
    }
  })
  return useQueries(queries);
}
```

**Comment by TkDodo** [comment]
there is no such thing as `useInfiniteQueries`, sorry.

**Reply by tchaton** [reply]
@TkDodo Any plan to add support for it ? This would be dope !

**Reply by tchaton** [reply]
I ended up re-writing an infinite query by hand.
